lib/core/src/sdk.rs

```diff
@@ -2771,6 +2770,22 @@ impl LiquidSdk {
                 // An unconfirmed tx that was not found in the payments table
                 self.persister.insert_or_update_payment_with_wallet_tx(tx)?;
             }
+            unconfirmed_txs_by_id.remove(&tx_id);
```

Comment:
So we are removing it if its still seen by the wallet, not necessarily confirmed?